File: Euclid.Storage/Database/Data/Navigator/NavigatorCategoryData.cs
using FluentNHibernate.Mapping;

namespace Euclid.Storage.Database.Data
{
    class NavigatorCategoryMapping : ClassMap<NavigatorCategoryData>
    {
        public NavigatorCategoryMapping()
        {
            Table("room_category");
            Id(x => x.Id, "id");
            Map(x => x.OrderId, "order_id");
            Map(x => x.ParentId, "parent_id");
            Map(x => x.Name, "name");
            Map(x => x.IsNode, "is_node");
            Map(x => x.VisibleRank, "visible_rank");
            Map(x => x.IsPublicSpaces, "is_public_spaces");
            Map(x => x.IsTradingAllowed, "is_trading_allowed");
        }
    }

    public class NavigatorCategoryData
    {
        public virtual int Id { get; set; }
        public virtual int OrderId { get; set; }
        public virtual int ParentId { get; set; }
        public virtual string Name { get; set; }
        public virtual int VisibleRank { get; set; }
        public virtual bool IsNode { get; set; }
        public virtual bool IsPublicSpaces { get; set; }
        public virtual bool IsTradingAllowed { get; set; }
    }
}

File: euclid-storage/src/main/java/euclid/storage/database/data/NavigatorCategoryData.java
package euclid.storage.database.data;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "room_category")
public class NavigatorCategoryData {

    @Id
    @Column(name = "id")
    private int id;

    @Column(name = "order_id")
    private int orderId;

    @Column(name = "parent_id")
    private int parentId;

    @Column(name = "name")
    private String name;

    @Column(name = "is_node")
    private boolean node;

    @Column(name = "visible_rank")
    private int visibleRank;

    @Column(name = "is_public_spaces")
    private boolean publicSpaces;

    @Column(name = "is_trading_allowed")
    private boolean tradingAllowed;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getOrderId() {
        return orderId;
    }

    public void setOrderId(int orderId) {
        this.orderId = orderId;
    }

    public int getParentId() {
        return parentId;
    }

    public void setParentId(int parentId) {
        this.parentId = parentId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isNode() {
        return node;
    }

    public void setNode(boolean node) {
        this.node = node;
    }

    public int getVisibleRank() {
        return visibleRank;
    }

    public void setVisibleRank(int visibleRank) {
        this.visibleRank = visibleRank;
    }

    public boolean isPublicSpaces() {
        return publicSpaces;
    }

    public void setPublicSpaces(boolean publicSpaces) {
        this.publicSpaces = publicSpaces;
    }

    public boolean isTradingAllowed() {
        return tradingAllowed;
    }

    public void setTradingAllowed(boolean tradingAllowed) {
        this.tradingAllowed = tradingAllowed;
    }
}
